src/ethereum-canister/tests/test_canister/mod.rs

```diff
@@ -0,0 +1,128 @@
+use candid::{utils::ArgumentEncoder, IDLArgs};
+use eyre::{ensure, Result, WrapErr};
+use temp_dir::TempDir;
+
+const DEFAULT_CONSENSUS_RPC: &str = "https://www.lightclientdata.org";
+const DEFAULT_EXECUTION_RPC: &str = "https://ethereum.publicnode.com";
+
+#[derive(Debug)]
+pub struct TestCanister {
+    name: String,
+    temp_dir: TempDir,
+}
+
+impl TestCanister {
+    pub fn deploy(name: &str) -> Self {
+        let temp_dir = TempDir::new().unwrap();
+
+        // setup the tempdir
+        make_symlink(&temp_dir, "src");
+        make_symlink(&temp_dir, "target");
+        make_symlink(&temp_dir, "Cargo.toml");
+        make_symlink(&temp_dir, "Cargo.lock");
+        make_symlink(&temp_dir, "dfx.json");
+
+        // deploy
+        let canister = TestCanister {
+            temp_dir,
+            name: name.to_owned(),
+        };
+        canister.run_dfx(&["deploy", name]).unwrap();
+
+        canister
+    }
+
+    pub fn setup_ethereum_canister() -> Self {
+        let canister = Self::deploy("ethereum_canister");
+        let request = interface::SetupRequest {
+            consensus_rpc_url: DEFAULT_CONSENSUS_RPC.to_owned(),
+            execution_rpc_url: DEFAULT_EXECUTION_RPC.to_owned(),
+        };
+        let _: () = call!(canister, "setup", request).unwrap();
+        canister
+    }
+
+    pub fn call(&self, method: &str, args: impl ArgumentEncoder) -> Result<Vec<u8>> {
+        // convert arguments into format understood by `dfx`
+        let args = candid::utils::encode_args(args).wrap_err("encoding args")?;
+        let args = IDLArgs::from_bytes(&args).wrap_err("decoding dfx args")?;
+
+        let stdout = self
+            .run_dfx(&["canister", "call", &self.name, method, &args.to_string()])
+            .wrap_err(format!("calling '{method} {args}'"))?;
+
+        // convert results from the format understood by `dfx`
+        // to the candid binary representation
+        // note: decoding here to the correct type is not possible as decoding
+        //       binds the result's lifetime to the lifetime of output
+        //       this is because decoding supports also reference types
+        let stdout = std::str::from_utf8(&stdout).wrap_err("decoding output")?;
+        let output: IDLArgs = stdout.parse().wrap_err("parsing output")?;
+        output.to_bytes().wrap_err("encoding to candid")
+    }
+
+    fn remove(&self) {
+        self.run_dfx(&["canister", "stop", &self.name])
+            .expect("Stopping failed");
+        self.run_dfx(&["canister", "delete", &self.name])
+            .expect("Deleting failed");
+    }
+
+    fn run_dfx(&self, args: &[&str]) -> Result<Vec<u8>> {
+        let output = std::process::Command::new("dfx")
```

Comment:
In general I prefer to import instead of writing a long paths. Can you make it `Command`? Also make `std::str` as `str`.

Reply:
done